File: java/test/ParenthesesTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ParenthesesTest {
    @Test
    public void one() {
        assertEquals("()", Parentheses.balance("()"));
    }

    @Test
    public void two() {
        assertEquals("a(b)c", Parentheses.balance("a(b)c"));
    }

    @Test
    public void three() {
        assertEquals("", Parentheses.balance(")("));
    }

    @Test
    public void four() {
        assertEquals("", Parentheses.balance("((((("));
    }

    @Test
    public void five() {
        assertEquals("()()", Parentheses.balance("(()()("));
    }

    @Test
    public void six() {
        assertEquals("(())", Parentheses.balance(")(())("));
    }

    @Test
    public void seven() {
        assertEquals("()()()", Parentheses.balance(")())(()()("));
    }
}

File: javascript/parentheses.test.js
const parentheses = require('./parentheses')

test('one', () => {
  expect(parentheses.balance('()')).toBe('()')
})

// test('two', () => {
//   expect(parentheses.balance('a(b)c')).toBe('a(b)c')
// })

test('three', () => {
  expect(parentheses.balance(')(')).toBe('')
})

test('four', () => {
  expect(parentheses.balance('(((((')).toBe('')
})

test('five', () => {
  expect(parentheses.balance('(()()(')).toBe('()()')
})

test('six', () => {
  expect(parentheses.balance(')(())(')).toBe('(())')
})

test('seven', () => {
  expect(parentheses.balance(')())(()()(')).toBe('()()()')
})
